src/data/transactions.tsx

```diff
@@ -54,6 +59,21 @@ export const useTranscations = () => {
     }
   };
 
+  const getCategories = async () => {
+    const [, setCategories] = useRecoilState(categoriesState);
+    const port = "http://localhost:3000";
```

Comment:
This is repeated more than once, take it out of these function and make it reusable